File: P1 2016.2/Questao(2)/java/entities/produtos/Chocolate.java
package entities.produtos;

public class Chocolate extends Produto implements Comestivel {
    private String tipo;

    public Chocolate(double preco, int id, String tipo) {
        super(preco, id);
        this.tipo = tipo;
    }

    public String getTipo() {
        return tipo;
    }

    public void desembalar() {
        System.out.println("Abrindo a embalagem do chocolate: " + getTipo());
    }
}

File: P1 2016.2/Questao(2)/python/entities/produtos/Chocolate.py
from .Produto import Produto
from .Comestivel import Comestivel


class Chocolate(Produto, Comestivel):
    def __init__(self, preco: float, id: int, tipo: str) -> None:
        super().__init__(preco, id)
        self.tipo = tipo

    def desembalar(self):
        print("Abrindo o chocolate do tipo:", self.tipo)

    def __str__(self) -> str:
        return f"Preco: {self.preco}, ID:{self.id}, Tipo:{self.tipo}"


